gateway-js/src/config.ts

```diff
@@ -36,30 +45,59 @@ export type Experimental_DidFailCompositionCallback = ({
   readonly compositionMetadata?: CompositionMetadata;
 }) => void;
 
-export interface Experimental_CompositionInfo {
+export interface ServiceDefinitionCompositionInfo {
   serviceDefinitions: ServiceDefinition[];
   schema: GraphQLSchema;
   compositionMetadata?: CompositionMetadata;
 }
 
+export interface CsdlCompositionInfo {
+  schema: GraphQLSchema;
+  compositionId: string;
+  csdl: string;
+}
+
+export type CompositionInfo =
+  | ServiceDefinitionCompositionInfo
+  | CsdlCompositionInfo;
+
 export type Experimental_DidUpdateCompositionCallback = (
-  currentConfig: Experimental_CompositionInfo,
-  previousConfig?: Experimental_CompositionInfo,
+  currentConfig: CompositionInfo,
+  previousConfig?: CompositionInfo,
 ) => void;
 
+export type CompositionUpdate = ServiceDefinitionUpdate | CsdlUpdate;
```

Comment:
Hmm, it feels like these two subtypes are drastically different... I wonder if they should get the same `experimental_updateServiceDefinitions` hook at all, or if you can define one hook for one kind and another hook for the other? It's experimental though so maybe unimportant

Reply:
Yeah, I think this is a good call. I do think this type still has value since these two types will both be handled within the same code path, but splitting the hooks simplifies this a bit.

Let me know your thoughts on 4cae6cb, which should resolve this.